### PR TITLE
tests: split github workflow into multiple jobs

### PR DESCRIPTION
**Pull request by connorjclark**
Splits into four jobs (with no dependencies):

* build
* unit
* smoke
* misc

Currently takes ~19 minutes to run our workflow. Now it takes ~10 minutes.

Also uploading `dist/` as a build artifact. Even though currently there are no dependent jobs requiring the usage of artifacts, making `dist/` downloadable may prove useful.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,23 +37,15 @@ jobs:
 
     - run: yarn --frozen-lockfile
     - run: yarn build-all
-    - run: yarn diff:sample-json
-    - run: yarn type-check
-    - run: yarn lint
-    - run: yarn test-proto # Run before unit-core because the roundtrip json is needed for proto tests.
 
     # Run tests that require headfull Chrome.
     - run: sudo apt-get install xvfb
```
Comment (connorjclark) [inline .github/workflows/ci.yml:42]: `build` stops being a great name for this job past this point. All the xvfb stuff could be moved to a new job. a couple of them require `dist` so it'd need to be dependent on `build` and download the artifact.

`smoke` will take so much longer than this job so it'd ndb, I prefer to keep it like this and not split the job.

File: lighthouse-cli/test/smokehouse/lighthouse-runners/bundle.js [hunk 2]
```diff
@@ -15,7 +15,9 @@ const ChromeLauncher = require('chrome-launcher');
 const ChromeProtocol = require('../../../../lighthouse-core/gather/connections/cri.js');
 
 // Load bundle, which creates a `global.runBundledLighthouse`.
-require('../../../../dist/lighthouse-dt-bundle.js');
+// Add empty string to circumvent TS, otherwise a clean `dist/` folder
+// would result in `yarn type-check` failing.
+require('../../../../dist/lighthouse-dt-bundle.js' + '');
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/lighthouse-runners/bundle.js:20]: :)

**Comment by paulirish** [comment]
looks great!

i would also like to bikeshed how to parallelize this. :)  how about

1. move all xvfb stuff and tests from build -> unit
1. move all of misc to the start of unit
1. i suppose you're expecting to have a separate job for the ToT chrome smokes?

@brendankenny dunno what you mean about the implicit dependencies... i'm not seeing a problem so far...

**Comment by brendankenny** [comment]
> dunno what you mean about the implicit dependencies
⬇️ 
> move all xvfb stuff and tests from build -> unit
would expose several of them, for instance :)

**Comment by brendankenny** [comment]
regardless of any more juggling, it isn't going to go any faster than the smoke tests unless they're broken up. There's also some weird low hanging fruit, like why is uploading to buildtracker so slow?

**Comment by paulirish** [comment]
> why is uploading to buildtracker so slow?

looked into it a bit ago. it's primarily the cost of running brotli against all our bundles

**Comment by paulirish** [comment]
> > dunno what you mean about the implicit dependencies
> ⬇️
> > move all xvfb stuff and tests from build -> unit
> 
> would expose several of them, for instance :)

ah just built file dependencies! okay great. https://github.com/GoogleChrome/lighthouse/pull/10994 has what i was thinkin there.

**Comment by connorjclark** [comment]
Merged the additional ToT smoke test and made a separate job for it.

https://github.com/GoogleChrome/lighthouse/actions/runs/140296254 logged 12m for everything

**Pull request closed without merging**